### PR TITLE
[WASM] Make TRANSPOSE support tensor less than 4D

### LINKED ISSUE #1186: [WASM] TRANSPOSE doesn't support non-4D tensor
**Test Env:**
Chromium Version: nightly build  79.0.3917.0(2e4d91c)
Platform: all

**Expected Result:**
Tests should pass 

**Actual Result:**
TRANSPOSE doesn't support non-4D tensor
`
TypeError: Incorrect number of tuple elements for array_int32_4: expected=4, actual=2`

**How to Reproduce:**
Visit:
https://brucedai.github.io/webnnt/test/issue_transpose_2D.html?prefer=fast

### PR DESCRIPTION
**Pull request by akineeic**
Fix #1186.

### CODE REVIEW

**Comment by akineeic** [comment]
The op [transpose](https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/lite/kernels/internal/reference/reference_ops.h#L3048-L3097) in TFlite supports tensor less than 4D, but the passed parameter [perm](https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/lite/kernels/internal/types.h#L1037-L1040)  must be an array of length 4. So the only thing we need to do is to extend perm‘s length to 4.

I have do the regression check and the result is passes: 844, failures: 336. (passes: 842, failures: 338 before)

**Comment by akineeic** [comment]
@huningxin @ibelem PTAL, thanks.

**Comment by akineeic** [comment]
@huningxin Could you please take a look since this link to an issue with label 'bug p2', thanks.

File: src/nn/wasm/PreparedModel.js [hunk 1]
```diff
@@ -1264,6 +1264,18 @@ export default class PreparedModel {
...
+          let extended_perm = [];
+
+          for (let i = 0; i < perm.length; ++i){
+            extended_perm[i] = perm[i];
+          }
+          for (let i = perm.length; i < 4; ++i){
+            extended_perm[i] = 0;
+          }
+          perm = extended_perm;
+        }
```
Comment (huningxin) [inline src/nn/wasm/PreparedModel.js:1277]: can we just use `perm`?
```
// Extend perm to length 4 by appending 0
for (let i = perm.length; i < 4; ++i) {
  perm.push(0);
}
```

Comment (huningxin) [reply src/nn/wasm/PreparedModel.js:1277]: @akineeic  any thoughts?

Comment (akineeic) [reply src/nn/wasm/PreparedModel.js:1277]: Sorry I see your comment just now, it's better in this way. I'll modify it.

**Comment by akineeic** [comment]
@huningxin  I have modified the PR according to your suggestion, please review again thanks !

File: src/nn/wasm/PreparedModel.js (hunk 1 above)
Comment (huningxin) [reply src/nn/wasm/PreparedModel.js:1270]: no worries. looks good now.

**Pull request merged**